tests/integration/src/test/java/org/mule/test/construct/LifecycleTrackerProcessor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.construct;
+
+import static org.mockito.Mockito.mock;
+import org.mule.functional.junit4.TestLegacyMessageBuilder;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.api.construct.FlowConstruct;
+import org.mule.runtime.core.api.construct.FlowConstructAware;
+import org.mule.runtime.core.api.processor.Processor;
+import org.mule.runtime.core.management.stats.ComponentStatistics;
+import org.mule.test.core.lifecycle.AbstractLifecycleTracker;
+
+public class LifecycleTrackerProcessor extends AbstractLifecycleTracker implements FlowConstructAware, Processor {
```

Comment:
Almost. The difference between them is that the one in unit test uses InternalMessage.builder directly. The other one uses TestLegacyMessageBuilder